python/src/iceberg/expressions/base.py

```diff
@@ -137,6 +74,30 @@ def __invert__(self) -> "BooleanExpression":
         ...
 
 
+class BoundPredicate(ABC):
+    def __init__(self, left, right):
+        """A concrete predicate must have a `left` and `right`"""
+        self._left = left
+        self._right = right
+
+    @abstractmethod
+    def eval(self, obj) -> bool:
+        """Evaluate the bound predicate"""
+        ...
```

Comment:
The new linter stuff recently added is complaining about this line. The `...` isn't needed if you have a docstring. Same thing a few lines down in the `bind` method.